Project2

GaxysGalactic opened the pull request:
Linked List & Iterators

![Results](https://user-images.githubusercontent.com/44332093/191050063-22918c28-b254-4b20-889e-d50b85726e81.PNG)

GaxysGalactic committed:
Part 1 - TSLList Class

+ Added the template class for SLList
+ Made member variables and constructors

GaxysGalactic committed:
Part 2 - Access

+ Added methods to allow for access to elements in the array

GaxysGalactic committed:
Part 3 - Size

+ Added methods related to Size checking.

GaxysGalactic committed:
Part 4 & 5 - Modifiers, Copy Constructor and Destructor

+ Added modifier functions
+ Added Copy Constructor
+ Added Destructor
+ Created private method refactor

GaxysGalactic committed:
Coding Standards and Efficiency

o Fixed some coding standards issues in initialization
o Made the Copy Constructor more efficient given the lack of iterators

GaxysGalactic committed:
Second Part of Project, Bugfixes

+ Added functions for lambda use
+ Added iterator
+ Bugtesting in main
o Fixed bug in insert

GaxysGalactic committed:
More Bugfixes and Assignment Operator

+ Added assignment operator for completeness
+ Added more debugging in main

GaxysGalactic committed:
Better naming conventions

o Changed the name of various parameters and such to be more readable and understandable

GaxysGalactic committed:
Final Tests

+ Added an option to insert at the end
+ Implemented the LinkedList Test into the cpp file
o Fixed some formatting

camodesigns commented on Projects/LinkedListIterator/LinkedListIterator/LinkedListIterator/SLList.cpp:
```diff
@@ -0,0 +1,178 @@
+	std::cout << "\nInserting Elements...!\n";
+	List.Insert("Is", 0);
+	List.Insert("?", 4);
+	List.Insert("Dumb", 2);
```
JD showed us a change for this part of the test, please implement the modificaction
List.Insert("Dumb",3)

camodesigns commented on Projects/LinkedListIterator/LinkedListIterator/LinkedListIterator/SLList.h:
```diff
@@ -0,0 +1,374 @@
+	}
+
+	//-------------------------------------------------------------------------------------------
+	TSLList(const T& FirstElement)
```
I think the name could be a NewElement since it is a generator and can be reused

camodesigns commented on Projects/LinkedListIterator/LinkedListIterator/LinkedListIterator/SLList.h:
```diff
@@ -0,0 +1,374 @@
+	//-------------------------------------------------------------------------------------------
+	//										UTILITY FUNCTIONS
+	//-------------------------------------------------------------------------------------------
+	FNode* GetNode(const int Index)
```
Great implementation of this utility feature

simonkytr replied:
you could start to implement the iterations with this function.

camodesigns reviewed (commented):
In general you have a great organization under code standards. the implementation of your utility function in the methods and functions seems appropriate to me, the use of constants, nodes and elements in private is adequate, very good work

simonkytr commented on Projects/LinkedListIterator/LinkedListIterator/LinkedListIterator/SLList.h:
```diff
@@ -0,0 +1,374 @@
+	{
+		FNode* Current = Head;
+		for (int i = 0; i < Size; i++)
+		{
```
you could already implement the iterators with this function something like 

`for (T& Temporal : *this)
		{
			if (Predictate(Temporal))
			{
				return &Temporal->Element;
			}
		}`

simonkytr commented on Projects/LinkedListIterator/LinkedListIterator/LinkedListIterator/SLList.h:
```diff
@@ -0,0 +1,374 @@
+	//										ITERATOR
+	//-------------------------------------------------------------------------------------------
+	class FIterator
+	{
```
you should reorder your class declaring the Iterator class to the beginning so you can use the iterators inside of your functions, maybe if you have the time could you prove it

simonkytr reviewed (changes requested):
Your work is always so interesting furthermore I learn a lot with your code, I made a few comments about the implementations for your methods, nothing unusual but pretty good,

GaxysGalactic committed:
Fixes to Code Review

o Made some small fixes according to the code review given. More will come later.
